lib/chef/provider/user/dscl.rb

```diff
@@ -239,8 +239,16 @@ def get_free_uid(search_limit=1000)
         #
         def uid_used?(uid)
           return false unless uid
-          users_uids = run_dscl("list /Users uid")
-          !! ( users_uids =~ Regexp.new("#{Regexp.escape(uid.to_s)}\n") )
+          users_uids = run_dscl("list /Users uid").split("\n")
+          uid_map = users_uids.inject({}) do |tmap, tuid|
+            x = tuid.split
+            tmap[x[1]] = x[0]
+            tmap
+          end
+          if uid_map[uid.to_s]
+            return true unless uid_map[uid.to_s] == @new_resource.username.to_s
+          end
+          return false
```

Comment:
So what is it you're looking for? This?

```
uid_map[uid.to_s] && uid_map[uid.to_s] != @new_resource.username.to_s
```
?

I think that's far less clear. How about:

```
if uid_map[uid.to_s]
  unless uid_map[uid.to_s] == @new_resource.username.to_s
    return true
  end
end
false
```